src/Plugin/Sampler/Bundle.php

```diff
@@ -203,10 +191,21 @@ protected function isBaseField(FieldDefinitionInterface $fieldConfig) {
   protected function getSupportedFieldDefinitions(string $entityTypeId, string $bundle) {
     $fieldDefinitions = $this->entityFieldManager->getFieldDefinitions($entityTypeId, $bundle);
     $supportedFieldTypes = array_flip($this->samplerSettings->get('supported_field_types'));
+    $supportedEntityTypes = array_flip($this->samplerSettings->get('supported_entity_types'));
+
+    $supportedFields = array_filter(
+      $fieldDefinitions,
+      function ($fieldDefinition) use ($supportedFieldTypes, $supportedEntityTypes) {
+        $fieldType = $fieldDefinition->getType();
+        $isSupported = isset($supportedFieldTypes[$fieldType]);
+
+        if ($isSupported && $this->isReferenceField($fieldDefinition)) {
+          $fieldDefinition->getSetting('target_type');
```

Comment:
You can remove this line - `$fieldDefinition->getSetting('target_type');`

In addition to that, I would suggest using an early return.
```
if (!isset($supportedFieldTypes[$fieldType])) {
  return FALSE;
}

if ($this->isReferenceField($fieldDefinition) && !isset($supportedEntityTypes[$fieldDefinition->getSetting('target_type')])) {
  return FALSE;
}
```

You can also use `empty` instead of `!isset`. 😉

Reply:
I like that